### PR TITLE
add kata-operator-payload dockerfiles

### PR DESCRIPTION
**Pull request by jensfr**
**- Description of the problem which is fixed/What is the use case**
It builds a container image with the kata-* and QEMU (plus
dependencies) RPMs in it. The kata-operator-daemon when running
on the worker nodes downloads this image and then installs
the RPMs it contains.

**- What I did**
This is a 1:1 copy from the release-4.6 branch
(commit 4765c70)

**- How to verify it**
The dockerfile can be used to build the container with podman build.
The build should finish succesfully. The payload image can be tested by
using the payload configmap to make the operator use it.

### CODE REVIEW

**Comment by bpradipt** [comment]
@jensfr looks like the related release-4.6 commit is `4765c70`

**Comment by jensfr** [comment]
@bpradipt @zanetworker could you please ack this?

File: images/payload/hack/prepare.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (zanetworker) [inline images/payload/hack/prepare.sh:1]: Where is `hack/prepare` used?

File: images/payload/hack/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+set -xe
+
+# Install the latest version of Docker
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
```
Comment (zanetworker) [inline images/payload/hack/prepare.sh:6]: Here you are assuming that the build will happen on linux, can you mention it somewhere?

File: images/payload/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# kata-operator-payload
+
+A container image used by kata-operator to install Kata and dependencies (like QEMU...) on worker nodes.
+During podman build the rpms are downloaded and a local repo is created. The resulting container image includes the rpms.
+The kata-operat-daemon pulls down this image unpacks it and installs it on the node where it is running.
```
[the remark this answers is not in the document]
Comment (zanetworker) [reply images/payload/README.md:5]: Also, I still see instances where `kata-install-daemon` is used, is this intended?

**Review by zanetworker — COMMENTED** [review]
I see you added a README which is awesome, what I saw missing though is how / where would one use the scripts introduced in this PR (e.g., prepare.sh, cleanup.sh, ...).

File: images/payload/hack/prepare.sh (hunk 2 above)
Comment (lysliu) [reply images/payload/hack/prepare.sh:6]: Where does `prepare.sh` referenced?

File: images/payload/hack/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+if "${TRAVIS_BRANCH}" == "master"; then
+	TAG="latest"
+else
+	TAG="$(echo ${TRAVIS_BRANCH} | awk '{print $2}' FS='-')"
+fi
+
+docker login \
+	--username="${QUAY_USERNAME}" \
+	--password="${QUAY_PASSWORD}" \
+	quay.io
```
[the remark this answers is not in the document]
Comment (jensfr) [reply images/payload/hack/deploy.sh:16]: No, I changed it to REGISTRY_USERNAME and _PASSWORD. Thank you!

File: images/payload/hack/prepare.sh (hunk 1 above)
Comment (jensfr) [reply images/payload/hack/prepare.sh:1]: I'm not sure. @fidencio ? Is prepare.sh called manually? Or is it somehow used by Travis?

**Comment by jensfr** [comment]
This is not required anymore because we are not using the payload image any more.

**Pull request closed without merging**